src/tests/functional/shared_test_classes/src/single_op/rdft.cpp

```diff
@@ -4,9 +4,59 @@
 
 #include "shared_test_classes/single_op/rdft.hpp"
 #include "common_test_utils/node_builders/rdft.hpp"
+#include "common_test_utils/ov_tensor_utils.hpp"
+#include "shared_test_classes/base/utils/ranges.hpp"
 
 namespace ov {
 namespace test {
+static inline void set_real_number_generation_data(utils::InputGenerateData& inGenData) {
+    inGenData.range = 8;
+    inGenData.resolution = 32;
+}
+
+void RDFTLayerTest::generate_inputs(const std::vector<ov::Shape>& targetInputStaticShapes) {
+    std::vector<size_t> input_shape;
+    ov::element::Type model_type;
+    std::vector<int64_t> axes;
+    std::vector<int64_t> signal_size;
+    ov::test::utils::DFTOpType op_type;
+    std::string target_device;
+    std::tie(input_shape, model_type, axes, signal_size, op_type, targetDevice) = this->GetParam();
+
+    auto elemType = model_type;
+
+    ov::test::utils::InputGenerateData inGenData;
+    if (elemType.is_real()) {
+        set_real_number_generation_data(inGenData);
+    }
+
+    const auto& funcInputs = function->inputs();
+    auto funcInput = funcInputs.begin();
+
+    const auto node = funcInput->get_node_shared_ptr();
+    const size_t inNodeCnt = node->get_input_size();
+
+    auto it = ov::test::utils::inputRanges.find(ov::op::v9::RDFT::get_type_info_static());
+    if (op_type == ov::test::utils::DFTOpType::INVERSE) {
+        it = ov::test::utils::inputRanges.find(ov::op::v9::IRDFT::get_type_info_static());
+    }
+
+    if (it != ov::test::utils::inputRanges.end()) {
+        const auto& ranges = it->second;
+        if (ranges.size() != 2) {
+            throw std::runtime_error("Incorrect size of ranges. It should be 2 (real and int cases)");
+        }
+        const auto& range = ranges.at(elemType.is_real());
+        inGenData = range.size() < inNodeCnt ? range.front() : range.at(0);
+    }
+
+    inputs.clear();
+    runtime::Tensor data_tensor = ov::test::utils::create_and_fill_tensor_act_dft(funcInput->get_element_type(),
+                                            targetInputStaticShapes[0],
+                                            inGenData.range, inGenData.start_from, inGenData.resolution, inGenData.seed);
+    inputs.insert({funcInput->get_node_shared_ptr(), data_tensor});
```

Comment:
Why custom `generate_inputs()` required?

Reply:
Yes. It's needed for fp16 precision. While 125993 does not implemented.